### PR TITLE
feat(autoware_launch): add param for configurable lateral_distance

### PR DESCRIPTION
**Pull request by beyzanurkaya**
Signed-off-by: beyza <bnk@leodrive.ai>

## Description

This PR adds parameters for using predicted objects to autoware_launch/config

## Related Link

https://github.com/autowarefoundation/autoware.universe/pull/2290

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_launch/rviz/autoware.rviz [hunk 1]
```diff
@@ -1614,7 +1594,11 @@ Visualization Manager:
                           Enabled: true
                           Name: ObstacleStop
                           Namespaces:
-                            {}
+                            detection_polygons: true
```
Comment (satoshi-ota) [inline autoware_launch/rviz/autoware.rviz:1597]: @beyzanurkaya You wanna add only this line, right? If so, could you clean up the diffs that are not related to this PR?

Comment (beyzanurkaya) [reply autoware_launch/rviz/autoware.rviz]: @satoshi-ota Yes you're right. I cleaned it.

Comment (satoshi-ota) [reply autoware_launch/rviz/autoware.rviz]: @beyzanurkaya Thanks. Now, merge ready?

File: autoware_launch/config/planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/obstacle_stop_planner.param.yaml [hunk 2]
```diff
@@ -7,17 +7,22 @@
...
     voxel_grid_z: 100000.0                   # voxel grid z parameter for filtering pointcloud [m]
+    use_predicted_objects: False            # whether to use predicted objects [-]
+    publish_obstacle_polygon: False          # whether to publish obstacle polygon [-]
 
     stop_planner:
       # params for stop position
       stop_position:
         max_longitudinal_margin: 5.0         # stop margin distance from obstacle on the path [m]
-        min_longitudinal_margin: 5.0         # stop margin distance when any other stop point is inserted in stop margin [m]
+        min_longitudinal_margin: 2.0         # stop margin distance when any other stop point is inserted in stop margin [m]
```
Comment (satoshi-ota) [inline autoware_launch/config/planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/obstacle_stop_planner.param.yaml:17]: this diff is related to this PR ?

File: autoware_launch/config/planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/obstacle_stop_planner.param.yaml [hunk 3]
```diff
@@ -26,13 +31,16 @@
       # params for slow down section
       slow_down_section:
         longitudinal_forward_margin: 5.0     # margin distance from slow down point to vehicle front [m]
-        longitudinal_backward_margin: 0.0    # margin distance from slow down point to vehicle rear [m]
+        longitudinal_backward_margin: 5.0    # margin distance from slow down point to vehicle rear [m]
```
Comment (satoshi-ota) [inline autoware_launch/config/planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/obstacle_stop_planner.param.yaml:34]: this diff is related to this PR ?

Comment (beyzanurkaya) [reply autoware_launch/config/planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/obstacle_stop_planner.param.yaml]: No, it isn't, I reverted it.

Comment (beyzanurkaya) [reply autoware_launch/config/planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/obstacle_stop_planner.param.yaml]: It' ready to merge

**Review by satoshi-ota — APPROVED** [review]
LGTM
